### PR TITLE
feat(FEC-9227): support reInit of MSE

### PR DESCRIPTION
**Pull request by Yuvalke**
### Description of the Changes

Destroy and initialize shaka before and after ads for now(should be resolved by shaka later and it'll be replace by attach detach media)

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/dash-adapter.js [hunk 1]
```diff
@@ -403,6 +408,43 @@ export default class DashAdapter extends BaseMediaSourceAdapter {
...
+  /**
+   * detach media - will remove the media source from handling the video
+   * @public
+   * @returns {void}
+   */
+  detachMediaSource(): void {
+    if (this._shaka) {
+      this._lastTimeDetach = this.currentTime;
+      this._reset().then(() => {
+        this._shaka = null;
```
Comment (OrenMe) [inline src/dash-adapter.js]: should probably go into `_reset`

Comment (Yuvalke) [reply src/dash-adapter.js]: I'm not sure, I'm using _reset on destroy as well and there it returns promise from destroy

File: src/dash-adapter.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 // @flow
 import shaka from 'shaka-player';
-import {AudioTrack, BaseMediaSourceAdapter, Error, EventType, TextTrack, Track, Utils, VideoTrack} from '@playkit-js/playkit-js';
+import {AudioTrack, BaseMediaSourceAdapter, Error, EventType, Html5EventType, TextTrack, Track, Utils, VideoTrack} from '@playkit-js/playkit-js';
```
Comment (yairans) [inline src/dash-adapter.js]: remove. take the event from `EventType`

File: src/dash-adapter.js [hunk 3]
```diff
@@ -403,6 +408,43 @@ export default class DashAdapter extends BaseMediaSourceAdapter {
     }
   }
 
+  /**
+   * attach media - return the media source to handle the video tag
+   * @public
+   * @param {boolean} playbackEnded don't seek to the last detach point
```
Comment (yairans) [inline src/dash-adapter.js]: fix

File: src/dash-adapter.js [hunk 4]
```diff
@@ -403,6 +408,43 @@ export default class DashAdapter extends BaseMediaSourceAdapter {
...
+   * @returns {void}
+   */
+  attachMediaSource(playbackEnded: ?boolean): void {
+    if (!this._shaka) {
+      if (this._videoElement && this._videoElement.src) {
+        Utils.Dom.setAttribute(this._videoElement, 'src', '');
+        Utils.Dom.removeAttribute(this._videoElement, 'src');
```
Comment (yairans) [inline src/dash-adapter.js]: ???

Comment (Yuvalke) [reply src/dash-adapter.js]: Case that ima set empty source and it raise an error from video tag

Comment (Yuvalke) [reply src/dash-adapter.js]: It could happened when IMA doesn't clear the src on ADBREAKEND or didn't finish the clear

File: src/dash-adapter.js [hunk 5]
```diff
@@ -289,6 +294,7 @@ export default class DashAdapter extends BaseMediaSourceAdapter {
     DashAdapter._logger.debug('Creating adapter. Shaka version: ' + shaka.Player.version);
     super(videoElement, source, config);
     this._setShakaConfig();
+    this._eventManager = new EventManager();
```
Comment (yairans) [inline src/dash-adapter.js]: move it to `BaseMediaSourceAdapter` for the all adapters

Comment (yairans) [reply src/dash-adapter.js]: and remove from hls then

**Pull request merged**